src/Tools/Source/RunTests/AssemblyScheduler.cs

```diff
@@ -2,359 +2,314 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
+using System.Collections;
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Diagnostics;
+using System.Diagnostics.Contracts;
 using System.IO;
 using System.Linq;
 using System.Reflection;
 using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
 using System.Reflection.PortableExecutable;
-using System.Text;
+using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.VisualStudio.Services.Common;
 
 namespace RunTests
 {
-    internal readonly struct AssemblyInfo
+    internal record struct WorkItemInfo(ImmutableSortedDictionary<AssemblyInfo, ImmutableArray<TestMethodInfo>> Filters, int PartitionIndex)
```

Comment:
@jaredpar / @jcouv  - happy to see if I can explicitly run certain assemblies by themselves, but in general running multiple together really helps with building few work items that run in similar amounts of time.

Also, besides the tests modified here, I haven't seen any new failures.  But it could just be because the partitioning happens to run them separately.  Could you tell me which assemblies / tests have this requirement?  I can check to see whats going on with them

Reply:
quick ping on this @jaredpar / @jcouv - could you point me to the assemblies which shouldn't be run together so I can ensure they always run by themselves?

  I'd also be curious as to why they need to be run separately - afaik vstest.console will spin up a different test process for each one

Reply:
@jcouv my concern here is the new assembly you added which has all of our stress / limit tests in it. One of the invariants you had for that is all the methods in the type need to run every time (to get the consistency we need). I've forgotten the name of that.

Reply:
Added a new assembly attribute (RunTestsInSinglePartitionAttribute) which can be used to mark that an assembly must run in its own work item.  Added the attribute to the MS.CA.CSharp.EndToEnd.UnitTests project